runtime/src/rent_collector.rs

```diff
@@ -30,21 +40,35 @@ impl RentCollector {
         epoch_schedule: &EpochSchedule,
         slots_per_year: f64,
         rent: &Rent,
+        operating_mode: OperatingMode,
     ) -> Self {
         Self {
             epoch,
             epoch_schedule: *epoch_schedule,
             slots_per_year,
             rent: *rent,
+            enable_old_behavior: Self::enable_old_behavior(epoch, operating_mode),
         }
     }
 
-    pub fn clone_with_epoch(&self, epoch: Epoch) -> Self {
+    pub fn clone_with_epoch(&self, epoch: Epoch, operating_mode: OperatingMode) -> Self {
         Self {
             epoch,
+            enable_old_behavior: Self::enable_old_behavior(epoch, operating_mode),
             ..self.clone()
         }
     }
+
+    fn enable_old_behavior(epoch: Epoch, operating_mode: OperatingMode) -> bool {
+        let enable_new_behavior = match operating_mode {
+            OperatingMode::Development => true,
+            OperatingMode::Preview => epoch >= Epoch::max_value(),
+            OperatingMode::Stable => epoch >= Epoch::max_value(),
+        };
+
+        !enable_new_behavior
```

Comment:
NIT: The `!`ing back and forth here and in the next hunk is a little clumsy to follow.  Maybe just settle on one? (new seems like it'd read best across the board)

Reply:
yeah, I've been bothered with some convoluted handling for gating... Now your mentioning this encouraged me to do the correct thing: https://github.com/solana-labs/solana/pull/11349/commits/047b43ab115d92937812e79ded7e0d3dddf0a58b. :)